### ISSUE TITLE
Veracode API ID and Key support

### ISSUE THREAD

**Issue by tjarrettveracode** [mort666/veracode-api#2]
Just a note that Veracode is deprecating username and password authentication for Veracode APIs. If this script is not updated to work with Veracode API ID and Key it will stop working once basic auth support is fully removed.

**Comment by mort666** [comment]
Don't have access to test or verify any changes with the Veracode platform at this time, is there a test API or public documentation so this can be updated?

**Comment by tjarrettveracode** [comment]
@mort666 - yes on documentation. There is a [public help center documenting our signing algorithm and providing code examples](https://help.veracode.com/reader/LMv_dtSHyb7iIxAQznC~9w/hn2qc_7fz3zFYV~e4ulRaQ). There are also other sample HMAC implementations on our [Github page.](https://github.com/veracode/Veracode-Community-Projects) 

Unfortunately none of the HMAC samples are in Ruby. I may be able to draft some contributors to the project to help out though.

**Comment by mort666** [comment]
Okay... I should be able to work something out.. Any way to test an implementation available?

**Comment by tjarrettveracode** [comment]
Yes, I think so. I'll contact you out of band to get that set up.

**Comment by mort666** [comment]
Cool, drop me an email to the one on my profile.. I'm going to take a look at this during this week.

**Comment by tjarrettveracode** [comment]
Sorry about the delay. You should have an invitation to the platform. I gave you an account with submitter and reviewer roles, and created an [app profile here](https://analysiscenter.veracode.com/auth/index.jsp#AppMetaData:14212:572047::::::) for testing.

**Comment by mort666** [comment]
So have got a basic auth thing done, but it doesn't seem to work, I was using the CURL example to test and all the components look right when I test it with the 'sample' details in that CURL example but when it comes to the live Key and ID it isn't working?

**Comment by tjarrettveracode** [comment]
Hi @mort666 -- unfortunately you won't be able to use ID and Key with the curl example. Part of what we've done with ID and Key is implement HMAC signing which Curl doesn't support natively. 

Someone's written up instructions on [how to get our HMAC signing to work with Curl and openssl](https://gist.github.com/m9aertner/7ae804a5297617456f81c8b5a3a9305b) which may be helpful in proceeding.

**Comment by mort666** [comment]
Yeah that is the one I've mimicked, to use to validate the creation, I've got it to work and match up each step until the final signature, but that example annoyingly doesn't include the complete signature value. 

Tho, I think I may have fixed my test code as I'd a typo in there on the last step which was affecting the signature generation.
